src/Microsoft.DotNet.Tools.Test/Program.cs

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.Extensions.Testing.Abstractions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Test
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            var app = new CommandLineApplication(false)
+            {
+                Name = "dotnet test",
+                FullName = ".NET Test Driver",
+                Description = "Test Driver for the .NET Platform"
+            };
+
+            app.HelpOption("-?|-h|--help");
+
+            var parentProcessIdOption = app.Option("--parentProcessId", "Parent process ID.", CommandOptionType.SingleValue);
+            var portOption = app.Option("--port", "Port number to listen for a connection.", CommandOptionType.SingleValue);
+            var projectPath = app.Argument("<PROJECT>", "The project to compile, defaults to the current directory. Can be a path to a project.json or a project directory.");
+            
+            app.OnExecute(() =>
+            {
+                try
+                {
+                    // Register for parent process's exit event
+                    if (parentProcessIdOption.HasValue())
+                    {
+                        int processId;
+
+                        if (!Int32.TryParse(parentProcessIdOption.Value(), out processId))
+                        {
+                            throw new InvalidOperationException($"Invalid process id '{parentProcessIdOption.Value()}'. Process id must be an integer.");
+                        }
+
+                        RegisterForParentProcessExit(processId);
+                    }
+                    
+                    var projectContexts = CreateProjectContexts(projectPath.Value);
+
+                    var projectContext = projectContexts.First();
+
+                    var testRunner = projectContext.ProjectFile.TestRunner;
+                    
+                    if (portOption.HasValue())
```

Comment:
I would wrap this if/else into its own method with a good name and push the decision on what to run to there. Even better would be to push this branching into its own abstraction. But I think a separate method is good enough.